From 76ba63f3cc55912e7d7886c0806640c5c6b726db Mon Sep 17 00:00:00 2001
From: Fabrice Bellet <fabrice@bellet.info>
Date: Wed, 25 Mar 2015 16:08:34 +0100
Subject: [PATCH 4/7] locator: Correct source accuracy comparison

We now stop sources whose accuracy level becomes higher, rather than
lower, than the requested level.

https://bugs.freedesktop.org/show_bug.cgi?id=89716
---
 src/gclue-locator.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/gclue-locator.c b/src/gclue-locator.c
index 42ce395..91df67e 100644
--- a/src/gclue-locator.c
+++ b/src/gclue-locator.c
@@ -184,7 +184,7 @@ on_avail_accuracy_level_changed (GObject    *gobject,
             !is_source_active (locator, src)) {
                 start_source (locator, src);
         } else if ((level == GCLUE_ACCURACY_LEVEL_NONE ||
-                    priv->accuracy_level > level) &&
+                    priv->accuracy_level < level) &&
                    is_source_active (locator, src)) {
                 g_signal_handlers_disconnect_by_func (G_OBJECT (src),
                                                       G_CALLBACK (on_location_changed),
-- 
2.1.0

